tests/unit/server/bundle_manager/make_bundles_test.py

```diff
@@ -0,0 +1,150 @@
+from codalab.worker.bundle_state import State
+from codalab.objects.dependency import Dependency
+from codalab.bundles.run_bundle import RunBundle
+from codalab.bundles.make_bundle import MakeBundle
+from codalab.bundles.dataset_bundle import DatasetBundle
+from codalab.lib.spec_util import generate_uuid
+import os
+import tempfile
+from tests.unit.server.bundle_manager import (
+    BaseBundleManagerTest,
+    BASE_METADATA,
+    BASE_METADATA_MAKE_BUNDLE,
+    BASE_METADATA_DATASET_BUNDLE,
+)
+
+
+class BundleManagerMakeBundlesTest(BaseBundleManagerTest):
+    def test_no_bundles(self):
+        """With no bundles available, nothing should happen."""
+        self.bundle_manager._make_bundles()
+        self.assertFalse(self.bundle_manager._is_making_bundles())
+
+    def test_restage_stuck_bundle(self):
+        """Bundles stuck in "MAKING" should be restaged and go back to the "MAKING" state."""
```

Comment:
When do bundles get stuck in MAKING?

Reply:
See this comment:

https://github.com/codalab/codalab-worksheets/blob/d50731b304eac68a39b5dbe1548a8bb6752c823f/codalab/server/bundle_manager.py#L175-L178